contracts/FlatLaunchpeg.sol

```diff
@@ -17,38 +17,37 @@ contract FlatLaunchpeg is BaseLaunchpeg, IFlatLaunchpeg {
     uint256 public override salePrice;
 
     /// @dev Emitted on initializePhases()
+    /// @param preMintStartTime Pre-mint start time in seconds
     /// @param allowlistStartTime Allowlist mint start time in seconds
     /// @param publicSaleStartTime Public sale start time in seconds
     /// @param publicSaleEndTime Public sale end time in seconds
     /// @param allowlistPrice Price of the allowlist sale in Avax
     /// @param salePrice Price of the public sale in Avax
     event Initialized(
+        uint256 preMintStartTime,
         uint256 allowlistStartTime,
         uint256 publicSaleStartTime,
         uint256 publicSaleEndTime,
         uint256 allowlistPrice,
         uint256 salePrice
     );
 
-    /// @dev Emitted on allowlistMint(), publicSaleMint()
-    /// @param sender The address that minted
-    /// @param quantity Amount of NFTs minted
-    /// @param price Price in AVAX for the NFTs
-    /// @param tokenId The token ID of the first minted NFT
-    event Mint(
-        address indexed sender,
-        uint256 quantity,
-        uint256 price,
-        uint256 tokenId
-    );
-
     modifier atPhase(Phase _phase) {
         if (currentPhase() != _phase) {
             revert Launchpeg__WrongPhase();
         }
         _;
     }
 
+    /// @dev Batch mint is allowed in the allowlist and public sale phases
+    modifier atBatchMintPhase() {
```

Comment:
nit: batch mint is not really a phase, can we say `isBatchMintCallable` or `isBatchMintAvailable` ?